docs/build-with-java.md

```diff
@@ -0,0 +1,261 @@
+## Build an IoT App with Java
+
+### Architecture
+
+In this section, we will walk through and explain the code for the different commands.
+As explained in the Getting Started page, the project is structured as follow:
+
+-   Migrate (`com.carepet.Migrate`) - creates the `carepet` keyspace and tables
+-   Collar (`com.carepet.Sensor`) - generates a pet health data and pushes it into the storage
+-   Web app (`com.carepet.server.App`) - REST API service for tracking pets health state
+
+### Migrate
+
+The `./bin/migrate.sh --hosts $NODE1 --datacenter datacenter1` command executes the main function in the `Migrate` class located `Migrate.java`. The function creates the keyspace and tables that will be used by the collar and server services.
+
+The below code in the `Migrate.java` file call the `createKeyspace` , `createSchema` then the `printMetadata` functions.
+
+```
+public static void main(String[] args) {
+	final Config config = Config.parse(new Config(), args);
+
+	final Migrate client = new Migrate(config);
+	client.createKeyspace();
+	client.createSchema();
+	client.printMetadata();
+}
+```
+
+There is a lot to unpack here. Let's break the code down line by line:
+
+The `config` object parses the arguments passed in the migrate command. In our case it's `hosts` and `datacenter`. The `hosts` argument expect the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you are using Scylla Cloud for instance. The command also accepts `username` and `password` arguments if required.
```

Comment:
```suggestion
The `config` object parses the arguments passed in the migrate command. In our case it's `hosts` and `datacenter`. The `hosts` argument expects the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you use Scylla Cloud. The command also accepts `username` and `password` arguments if required.
```

Reply:
A small update to Tzach's comment:
The `config` object parses the arguments passed in the `migrate command`. In this case, the arguments are `hosts` and `datacenter`. The `hosts` argument expects the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you use Scylla Cloud. The command also accepts `username` and `password` arguments if required.